plover/command/set_config.py

```diff
@@ -0,0 +1,29 @@
+from plover import log
+
+
+def set_config(engine, cmdline):
+    """
+    Set a Plover config option to a given value upon executing a stroke pattern.
+
+    Example usage (enter these lines into dictionary):
+    "O*EP": "{PLOVER:SET_CONFIG:translation_frame_opacity,100}",
+    "TR*PB": "{PLOVER:SET_CONFIG:translation_frame_opacity,0}",
+    """
+    # Break down the command string into a key and a value. Throw an error if it isn't in the form of a 2-tuple.
+    try:
+        key, value = tuple(cmdline.split(','))
+    except ValueError:
+        log.error("SET_CONFIG: Bad command string. Command must be in the form {PLOVER:SET_CONFIG:option,value}")
+        return
+    # We need to look up the old value of the config setting in order to determine its type.
+    try:
+        oldvalue = engine[key]
+    except KeyError:
+        log.error("SET_CONFIG: %s is not a valid Plover config option." % key)
```

Comment:
It's better to not catch the exception: the engine already does that, and this will ensure the rest of the translation (if any) is not sent (as it might depend on the command being successful). You could catch and re-raise if you do want to ensure a more meaningful error message.

Reply:
I just thought of something; should there be an extra level of exception messages for plugin commands? So that the user knows that the error is descended from the plugin and not the Plover engine itself?

Something like this:

```
def _consume_engine_command(self, command):
    ...
    command_args = command.split(':', 1)
    command_fn = registry.get_plugin('command', command_args[0]).obj
    try:
        command_fn(self, command_args[1] if len(command_args) == 2 else '')
    except Exception as e:
        raise Exception('Command error: %s' % command_args[0]) from e
```

I wouldn't have to manually add "SET_CONFIG" to each error message on the plugin side then.